hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java

```diff
@@ -147,10 +149,10 @@ public static MetadataStore getMetadataStore(FileSystem fs)
    */
   @RetryTranslated
   public static S3AFileStatus putAndReturn(MetadataStore ms,
-      S3AFileStatus status,
-      S3AInstrumentation instrumentation) throws IOException {
+      S3AFileStatus status, S3AInstrumentation instrumentation,
+      ITtlTimeProvider timeProvider) throws IOException {
```

Comment:
nit; once you go one arg per line (it scales well), don't merge back again...so put instrumentation back on its own line.